### PR TITLE
Feat/component masthead

### PR DESCRIPTION
**Pull request by harrisazmi**

### CODE REVIEW

File: apps/storybook/stories/masthead.stories.tsx
Comment (irfancoder) [inline apps/storybook/stories/masthead.stories.tsx:1]: 1. Missing dark mode story

File: packages/react/src/components/masthead.tsx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+}: PropsWithChildren) => {
+  return (
+    <div className="flex select-none items-center gap-2">
+      <MalaysiaFlagIcon className="shrink-0" />
+      <span className="text-txt-black-900">{children}</span>
+    </div>
+  );
+};
+
+export const MastheadIdentificationToggle = ({
```
Comment (irfancoder) [inline packages/react/src/components/masthead.tsx:44]: Rename `MastheadToggle`

File: packages/react/src/components/masthead.tsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+type MastheadSectionTitleProps = {
+  children: React.ReactNode;
+};
+
+type MastheadSectionBodyProps = {
+  children: React.ReactNode;
+};
+
+// Component definitions
+export const MastheadHeader = ({ children }: PropsWithChildren) => {
```
Comment (irfancoder) [inline packages/react/src/components/masthead.tsx:23]: 1. use `FunctionComponent<...>` generic when defining components.
2. Follow this pattern;
```tsx
// basic component ✅
const ComponentA: FunctionComponent<...> = ({ ... }) => <CompA />

// forwardRef component ✅
const ComponentB: ForwardRefExoticComponent<...> = (({ ... }, ref) => <CompB />)

// exports ✅
export { ComponentA, ComponentB }

// export + component  ❌
export const ComponentC = () => ...
```

File: packages/react/src/components/masthead.tsx (hunk 1 above)
Comment (irfancoder) [reply packages/react/src/components/masthead.tsx:44]: Does not highlight when tabbed (in focus)

File: packages/react/src/components/masthead.tsx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      <div className="gap-4.5 pt-4.5 grid grid-cols-1 pb-6 pl-6 sm:grid-cols-2 sm:gap-6 sm:pb-8 sm:pt-6">
+        <span className="text-primary-600 static text-sm sm:hidden">
+          Here's how you know
+        </span>
+        {children}
+      </div>
+    </div>
+  );
+};
+
```
Comment (irfancoder) [inline packages/react/src/components/masthead.tsx:67]: No transition?

File: packages/react/src/components/masthead.tsx [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+export const MastheadSection = ({ icon, children }: MastheadContentProps) => {
+  return (
+    <div className="flex gap-3">
+      <div className="text-txt-black-500 shrink-0">{icon}</div>
```
Comment (irfancoder) [inline packages/react/src/components/masthead.tsx:71]: The icon's position is not following the spec on figma

**Review by irfancoder — CHANGES_REQUESTED** [review]

File: apps/storybook/stories/masthead.stories.tsx
Comment (harrisazmi) [reply apps/storybook/stories/masthead.stories.tsx:1]: roger. adding it

File: packages/react/src/components/masthead.tsx (hunk 1 above)
Comment (harrisazmi) [reply packages/react/src/components/masthead.tsx]: fixed

File: packages/react/src/components/masthead.tsx (hunk 2 above)
Comment (harrisazmi) [reply packages/react/src/components/masthead.tsx:23]: followed the pattern as described

File: packages/react/src/components/masthead.tsx (hunk 3 above)
Comment (harrisazmi) [reply packages/react/src/components/masthead.tsx]: done boss

**Review by irfancoder — APPROVED** [review]

**Pull request merged**